### PR TITLE
Add Object Store credential support

### PR DESCRIPTION
**Pull request by haardikdharma10**
## Description

This PR adds support for Object Store Credentials and tests for objectstore.

### CODE REVIEW

File: objectstore_credential.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	AccountID         string  `json:"accountID"`
+	AcessKeyID        *string `json:"accessKeyID,omitempty"`
+	SecretAccessKeyID *string `json:"secretAccessKeyID,omitempty"`
+	MaxSizeGB         int     `json:"maxSize,omitempty"`
+	Suspended         bool    `json:"suspended"`
+	Status            string  `json:"status"`
+	Namespace         string  `json:"-"`
+}
+
+// PaginatedObjectstores is a paginated list of Objectstores
```
Comment (RealHarshThakur) [inline objectstore_credential.go:23]: Can you please fix the linter errors?

**Review by alejandrojnm — APPROVED** [review]
LGTM

File: objectstore_credential.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+type ObjectStoreCredential struct {
+	ID                string  `json:"id"`
+	Name              string  `json:"name"`
+	AccountID         string  `json:"accountID"`
+	AcessKeyID        *string `json:"accessKeyID,omitempty"`
+	SecretAccessKeyID *string `json:"secretAccessKeyID,omitempty"`
+	MaxSizeGB         int     `json:"maxSize,omitempty"`
+	Suspended         bool    `json:"suspended"`
+	Status            string  `json:"status"`
+	Namespace         string  `json:"-"`
```
Comment (RealHarshThakur) [inline objectstore_credential.go:20]: I don't think this should be here

File: objectstore_credential.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// ObjectStoreCredential holds the credential of an object store
+type ObjectStoreCredential struct {
+	ID                string  `json:"id"`
+	Name              string  `json:"name"`
+	AccountID         string  `json:"accountID"`
```
Comment (RealHarshThakur) [inline objectstore_credential.go:14]: We don't need account as a struct field.

**Review by RealHarshThakur — APPROVED** [review]

**Pull request merged**